db/fixtures/ae_datastore/ManageIQ/Service/Provisioning/StateMachines/Methods.class/__methods__/catalogiteminitialization.rb

```diff
@@ -81,17 +81,19 @@ def assign_service_tag(tag_category, tag_value)
   @service.tag_assign("#{tag_category}/#{tag_value}")
 end
 
-def get_vm_name(dialog_options_hash, prov)
-  log_and_update_message(:info, "Processing get_vm_name", true)
+def set_vm_name(dialog_options_hash, prov)
+  log_and_update_message(:info, "Processing set_vm_name", true)
   new_vm_name = dialog_options_hash.fetch(:vm_name, nil) || dialog_options_hash.fetch(:vm_target_name, nil)
-
-  new_vm_name = prov.get_option(:vm_target_name) if new_vm_name.blank?
-
-  dialog_options_hash[:vm_target_name] = new_vm_name
-  dialog_options_hash[:vm_target_hostname] = new_vm_name
-  dialog_options_hash[:vm_name] = new_vm_name
-  dialog_options_hash[:linux_host_name] = new_vm_name
-  log_and_update_message(:info, "Processing get_vm_name...Complete", true)
+  if new_vm_name.blank?
+    log_and_update_message(:info, "Using default vm name: #{prov.get_option(:vm_target_name)}", true)
+  else
```

Comment:
@mkanoor 
Shouldn't we set the new_vm_name in the else to this value:
new_vm_name = prov.get_option(:vm_target_name)